### DISCUSSION TITLE
Set jpeg quality?

### DISCUSSION THREAD

**Discussion by iansmith42** [gtxaspec/wz_mini_hacks#437] [Q&A]
Is there a way to change the default jpeg quality from 32 using cmd?  

Or can I get a bitmap from the camera and save it myself?

**Comment by iansmith42** [comment]
Also the same with the video stream.  I'm saving to my NAS, I'd love to bump up the bitstream for better quality.  The compression is pretty awful with a terrible lack of detail.  Any way to change it or is it all hard coded into their firmware?

**Reply by Bob-AOL** [reply]
You can consistently achieve 20~22fps at ~250kB/s using the USB/ETH feature.

**Reply by iansmith42** [reply]
I'm not sure I understand?  The camera is writing files to the SD card, what would changing to a USB ethernet adapter do?  It would also not affect JPEG stills.

Is it possible to access the cameras sensor directly?  The video I expect is not going to improve much due to CPU limitations but I don't see why getting high quality JPEGs out of it shouldn't be possible.  I poked around a bit but was unable to see any obvious way of grabbing data from the sensor.

I'd really like something better than JPEGS at quality 30!

**Reply by Bob-AOL** [reply]
I was responding to your streaming post, not the .jpg issue, though a better stream could indeed improve your .jpg quality by default should you implement a software solution on a machine with a processor capable of achieving both. Blue Iris obviously comes to mind, but there are other options as well.

The ethernet adapter would facilitate the best possible performance available from these cameras unless you have one helluva robust wi-fi network designed and dedicated specifically for/to them.

**Comment by iansmith42** [comment]
I'm not talking about extracting JPEGs from a network stream, and replacing the CPU is a little beyond my abilities. :)

It looks like you can get a raw capture using "echo saveraw 1 >/proc/jz/isp/isp-w02" but I'm going to have to figure out how to cross compile for this camera if I want anything easier to work with than a 4MB RAW file.  But good to know it's possible.

**Reply by Bob-AOL** [reply]
Way too much work. Capturing from a high quality stream is much simpler -- depending on your purpose, of course.

What do you use as your preferred viewer? And are you sure you're not possibly capturing in SD mode somehow?

**Reply by Bob-AOL** [reply]
And is this intended as a trigger activated automated capture process?

**Comment by iansmith42** [comment]
It's automated.  I can get the raw bayer-encoded data but am unable to covert it properly.  I'm using dcraw but I get a very pink and pixelated image out of it.

Filename: snap0.raw
Timestamp: Thu Jan  1 00:00:00 1970
Camera: Photron BC2-HD
ISO speed: 0
Shutter: 0.0 sec
Aperture: f/0.0
Focal length: 0.0 mm
Embedded ICC profile: no
Number of raw images: 1
Full size:   1920 x 1080
Image size:  1920 x 1080
Output size: 1920 x 1080
Raw colors: 3
Filter pattern: GB/RG
Daylight multipliers: 1.089011 0.973739 1.118123

It seems to be able to identify the format but for some reason is not converting it correctly.

Loading Photron BC2-HD image from snap0.raw ...
Scaling with darkness 0, saturation 65535, and
multipliers 1.118380 1.000000 1.148278 1.000000
AHD interpolation...
Converting to sRGB colorspace...
Writing data to snap0.ppm ...

Anyone have experience in converting that snap0.raw file?

**Reply by Bob-AOL** [reply]
Sorta beyond the scope of this venue. That said, curiosity got the best of me and I ran across this in fairly short order. May give you an idea or two that could apply to today's applications. https://stackoverflow.com/questions/56587645/how-to-convert-colored-image-from-raw-to-jpeg-format

Otherwise, I'd be researching alternative streaming software solutions for purposes of directly snagging jpegs from triggered events to your external destination. Expecting the camera firmware itself to do it directly to the SD card at an acceptable level would seem to be a futile exercise according to your own description of the results. 

Good luck.

**Reply by Bob-AOL** [reply]
More curiosity...

Snapshots from two of my v3s using the app on my Android phone. They both are obviously capable of rendering relatively decent static images in .jpeg format. A little washed out and kept at their original size. Only so many pixels available to work with...

![7C78B272686D_1672329194345](https://user-images.githubusercontent.com/110629141/209978730-1077f07e-226d-4514-be61-6ca860b9db44.jpg)

![D03F278A6A10_1672328982753](https://user-images.githubusercontent.com/110629141/209978829-f584acd5-13a8-47e4-83b3-1919c293aa64.jpg)

**Comment by iansmith42** [comment]
I think figuring out how to work with the cameras output is certainly within the venue. :)

I'm simply trying to get a still image that isn't compressed at a terrible 32 jpeg quality.  The built in jpeg routine over-compresses them badly.  I can get raw images but haven't yet figured out how to convert them yet.

Here is an example, look at the banding in the snow and the sky.

Someone HAS to know how to do this, that saveraw command is documented, so I just need to know how to process the output.  

![WyseV3](https://user-images.githubusercontent.com/44298052/210112138-5d841546-8953-42a3-8d33-0e0b8f1edc93.jpg)
